### PR TITLE
switch git to kernel.org source and sha256

### PR DESCRIPTION
**Pull request by healther**
sha256 source here: https://mirrors.edge.kernel.org/pub/software/scm/git/sha256sums.asc 

edit: Maybe I should add a bit of an explanation here. The idea for this PR started, when I added a newer git version and the sha256 that `spack` got wasn't the one from the signed document linked above. kernel.org is kind of anal regarding chain of trust, so I'd prefer using their infrastructure as much as possible rather than github.com's autogenerated tar-balls (where they don't guarantee checksum stability).
I basically replaced all md5sums into sha256s and moved the url to kernel.org

### CODE REVIEW

**Comment by healther** [comment]
One second thought: We could actually think about setting up a script to fetch new versions (and it should be perfectly save as we can just authenticate `sha256sums.asc` in the script)

File: var/spack/repos/builtin/packages/git/package.py [hunk 1]
```diff
@@ -34,129 +34,131 @@ class Git(AutotoolsPackage):
...
-            'md5': 'e04bfbbe5f17a4faa9507c75b8505c13',
-            'md5_manpages': 'f1d5dfc1459c9f2885f790c5af7473d1'
+            'sha256': 'e04bfbbe5f17a4faa9507c75b8505c13',
+            'sha256_manpages': 'f1d5dfc1459c9f2885f790c5af7473d1'
         },
         {
             'version': '2.17.0',
-            'md5': '8e0f5253eef3abeb76bd9c55386d3bee',
-            'md5_manpages': '1ce1ae78a559032810af8b455535935f'
+            'sha256': '8e0f5253eef3abeb76bd9c55386d3bee',
```
[the remark this answers is not in the document]
Comment (healther) [reply var/spack/repos/builtin/packages/git/package.py]: oops, sry

**Comment by healther** [comment]
@tgamblin this should address all your comments, if not feel free to ping me

**Comment by healther** [comment]
@tgamblin ping or maybe @adamjstewart

File: var/spack/repos/builtin/packages/git/package.py [hunk 2]
```diff
@@ -34,129 +34,131 @@ class Git(AutotoolsPackage):
     """
 
     homepage = "http://git-scm.com"
-    url      = "https://github.com/git/git/archive/v2.12.0.tar.gz"
+    url      = "https://mirrors.edge.kernel.org/pub/software/scm/git/git-2.12.0.tar.gz"
+    # url      = "https://github.com/git/git/archive/v2.12.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/git/package.py]: Can you delete this line?

Comment (healther) [reply var/spack/repos/builtin/packages/git/package.py]: done

**Pull request merged**